assets/pages/new-contributions-landing/contributionsLandingActions.js

```diff
@@ -193,10 +179,43 @@ const sendFormSubmitEventForPayPalRecurring = () =>
     onFormSubmit(formSubmitParameters);
   };
 
-function setValueAndTogglePayPal<T>(setStateValue: T => Action, value: T) {
-  return (dispatch: Function): void => {
+
+function enableOrDisableForm(formIsValid: boolean, state: State, dispatch: Function) {
+
+  const shouldEnable = checkoutFormShouldSubmit({
+    formValid: formIsValid,
+    canContributeWithoutSigningIn: canContributeWithoutSigningIn(
+      state.page.form.contributionType,
+      state.page.user.isSignedIn,
+      state.page.form.userTypeFromIdentityResponse,
+    ),
+    isRecurringContributor: state.page.user.isRecurringContributor,
+  });
+
+  if (shouldEnable && window.enablePayPalButton) {
+    dispatch(setFormIsSubmittable(true));
+    window.enablePayPalButton();
+  } else if (window.disablePayPalButton) {
+    dispatch(setFormIsSubmittable(false));
+    window.disablePayPalButton();
+  }
+}
+
+/*
+  Set the value of a form variable, then set the form validity and submitability
+*/
+function setValue<T>(setStateValue: T => Action, value: T) {
+  return (dispatch: Function, getState: () => State): void => {
+
     dispatch(setStateValue(value));
-    dispatch(togglePayPalButton());
+
+    const state = getState();
+
+    // TODO: use the actual form state rather than re-fetching from DOM
+    const formIsValid = formElementIsValid(getForm('form--contribution'));
+
+    dispatch(setFormIsValid(formIsValid));
```

Comment:
In a future PR, I am going to get rid of the setFormIsValid action, and make `setFormIsSubmittable` take a list of reasons (which can be the form is invalid, or they are an existing recurring contributor, or we are waiting on finding out their User Type from identity, etc).

Reply:
This will lessen the number of side effects here, and centralise/simplify the error message display logic, which I think would be nice. I like the idea of the complicated logic happening in once place (here), then the individual components not having to do too much work to figure out what do display.

Reply:
Great idea!